packages/node_modules/pouchdb-core/src/constructor.js

```diff
@@ -24,11 +24,24 @@ function prepareForDestruction(self) {
   }
 
   function onConstructorDestroyed() {
-    self.removeListener('destroyed', onDestroyed);
+    onClosed();
     self.emit('destroyed', self);
   }
 
+  function onClosed() {
+    self.removeListener('destroyed', onDestroyed);
```

Comment:
Conversely I don't understand why we say `self.emit('destroyed',self)` on line 28, instead of simply `self.emit('destroyed')` (the code in this master). The current only consumer is the replication code, which doesn't handle the parameter; and the parameter looks redundant in itself.